recipes/xsd/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.40.0"
+
+
+class ConanXqilla(ConanFile):
+    name = "xsd"
+    
+    description = (
+        "XSD is a W3C XML Schema to C++ translator. It generates vocabulary-specific, statically-typed C++ mappings (also called bindings) from XML Schema definitions. XSD supports two C++ mappings: in-memory C++/Tree and event-driven C++/Parser."
+    )
+    topics = ("xsd", "xml", "c++")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://codesynthesis.com/projects/xsd/"
+    license = ("GPL-2.0","FLOSSE")
+    settings = "os", "arch", "compiler"
+
+    exports_sources = "patches/**"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("xerces-c/3.2.3")
+
+    @property
+    def _doc_folder(self):
+        return os.path.join(self._source_subfolder,"xsd","doc")
+
+    @property
+    def _make_cmd(self):
+        return self._gnumake_cmd
+
+    @property
+    def _make_program(self):
+        return tools.get_env('CONAN_MAKE_PROGRAM', tools.which('make'))
+    
+    @property
+    def _gnumake_cmd(self):
+        make_ldflags = "LDFLAGS='{libs} -pthread'".format(
+            libs=" ".join(["-L{}".format(os.path.join(self.deps_cpp_info["xerces-c"].rootpath, it)) for it in self.deps_cpp_info["xerces-c"].libdirs]))
+        flags = []
+        flags.append(' '.join(["-I{}".format(os.path.join(self.deps_cpp_info["xerces-c"].rootpath, it)) for it in self.deps_cpp_info["xerces-c"].includedirs]))
+        if self.settings.compiler == "gcc":
+            flags.append('-std=c++11')
+        make_ccpflags = "CPPFLAGS='{}'".format(" ".join(flags))
+        make_cmd = '{ldflags} {cppflags} {make} -j{cpucount}'.format(
+            make=self._make_program, ldflags=make_ldflags, cppflags=make_ccpflags,cpucount=tools.cpu_count())
+        return make_cmd
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("The xsd recipe currently only supports Linux.")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
```

Comment:
To avoid building the binary for each compiler.
```suggestion
            tools.check_min_cppstd(self, 11)

    def package_id(self):
        del self.info.settings.compiler

    def source(self):
```

Reply:
This is particularly important
```
    def package_id(self):
        del self.info.settings.compiler
 ```
 Otherwise you have to consume the package using the same compiler, which shouldn't be enforced.
 What I mean is, I should be able to pull in a tool compiled with gcc 10 even if my project is being compiled with gcc 8.

Reply:
Add package_id